src/Castle.Core/DynamicProxy/IProxyBuilderWithCache.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2004-2021 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.DynamicProxy
+{
+	using System.Reflection;
+
+	using Castle.DynamicProxy.Serialization;
+
+	/// <summary>
+	///   A <see cref="IProxyBuilder"/> that caches generated proxy types for reuse.
+	/// </summary>
+	public interface IProxyBuilderWithCache : IProxyBuilder
+	{
+#if FEATURE_SERIALIZATION
```

Comment:
Due to the ordering of this `#if`, targets that don't support full binary serialization still get this interface (but it will be empty). I chose this&mdash;rather than erasing the interface as a whole&mdash;because it is true on all targets to say that "`DefaultProxyBuilder` is a builder with a cache", even when you cannot batch-populate it. It also gives us more API uniformity, as `DefaultProxyBuilder` will implement the same interfaces for all targets.